src/module_2/groceries_analysis.md

```diff
@@ -0,0 +1,1811 @@
+    </tr>
+    <tr>
+      <th>4694</th>
+      <td>085d8e598139ce6fc9f75d9de97960fa9e1457b409ec00...</td>
+      <td>Top Up</td>
+      <td>UKJ</td>
+      <td>2021-11-02 13:50:06</td>
+      <td>2021-11-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+    <class 'pandas.core.frame.DataFrame'>
+    Index: 4983 entries, 2160 to 3360
+    Data columns (total 10 columns):
+     #   Column                 Non-Null Count  Dtype  
+    ---  ------                 --------------  -----  
+     0   user_id                4983 non-null   object 
+     1   user_segment           4983 non-null   object 
+     2   user_nuts1             4932 non-null   object 
+     3   first_ordered_at       4983 non-null   object 
+     4   customer_cohort_month  4983 non-null   object 
+     5   count_people           325 non-null    float64
+     6   count_adults           325 non-null    float64
+     7   count_children         325 non-null    float64
+     8   count_babies           325 non-null    float64
+     9   count_pets             325 non-null    float64
+    dtypes: float64(5), object(5)
+    memory usage: 428.2+ KB
+    
+    Mising values(NAs):
+
+
+
+    user_id                     0
+    user_segment                0
+    user_nuts1                 51
+    first_ordered_at            0
+    customer_cohort_month       0
+    count_people             4658
+    count_adults             4658
+    count_children           4658
+    count_babies             4658
+    count_pets               4658
+    dtype: int64
+
+
+## Check Orders 
+
+With the orders dataframe, we could try to identify the purchasing patterns of customers. Both, the number of orders per user and the number of items per order are important metrics to understand the purchasing patterns of customers. 
+
+
+
+Let's first check the frequency of orders per user.
+
+
+```python
+orders_frequency = orders_df['user_id'].value_counts().reset_index().rename(columns={'count': 'order_count'})
+display(orders_frequency['order_count'].describe())
+plt.hist(orders_frequency["order_count"],bins = len(orders_frequency["order_count"].unique()), edgecolor="k")
+plt.title('Distribution of Order Frequencies per User')
+plt.xlabel('Number of Orders')
+plt.ylabel('Number of Users')
+plt.xlim(min(orders_frequency["order_count"]-1), max(orders_frequency["order_count"]))
+plt.grid(axis='y', alpha=0.75)
+plt.show()
+
+```
+
+
+    count    4983.000000
+    mean        1.760586
+    std         1.936537
+    min         1.000000
+    25%         1.000000
+    50%         1.000000
+    75%         2.000000
+    max        25.000000
+    Name: order_count, dtype: float64
+
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_9_1.png)
+    
+
+
+Most of users just make 1 order, which indicates that the platform has a large base of "one time" users. Let's now check the distribution of order sizes.
```

Comment:
Nice here that you do have a conclusive cell. I'd oberve that it seems that there is a fat tail in the distribution: a handfull of users with many orders. This is a common theme in real life